course/src/isometry.cc

```diff
@@ -0,0 +1,131 @@
+#include <cmath>
+#include <cstdint>
+#include "isometry.h"
+
+namespace ekumen {
+namespace math {
+
+double& Vector3::operator [] (const int index) const {
+    if(index > Vector3::MAX_INDEX) {
+        //throw error
+    }
+    return *(data_[index]);
```

Comment:
Ohhh!!! This is how you're sidestepping [this](https://github.com/LucasJSch/cppl1_q12020/pull/1/files#diff-dfb4d63ae36c57de208059ac43d21756R23)! You managed to fool the compiler, but this will definitely bite you back sometime in the future when you use this.

This is dangerous. The correct way is to make this overload return a `const double&` instead of a `double &`. And don't go back and forth between pointers and variables. The compiler loses track of the cv status of the things you're pointing at.